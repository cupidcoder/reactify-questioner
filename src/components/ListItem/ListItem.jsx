import React from 'react';
import { Link } from 'react-router-dom';
import style from './ListItem.css';

const ListItem = ({ text, path, clicked }) => (
  <li className={style.ListItem}>
    <Link to={path || '#'} onClick={clicked}>
      {text}
    </Link>
  </li>
);

export default ListItem;
