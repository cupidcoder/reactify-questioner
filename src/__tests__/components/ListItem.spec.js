import React from 'react';
import { shallow } from 'enzyme';
import ListItem from '../../components/ListItem/ListItem';

describe('ListItem', () => {
  it('Should correctly save snapshot of the ListItem', () => {
    const items = {
      text: 'test',
      path: '/somewhere'
    };
    const wrapper = shallow(<ListItem {...items} />);
    expect(wrapper).toMatchSnapshot();
  });
});
