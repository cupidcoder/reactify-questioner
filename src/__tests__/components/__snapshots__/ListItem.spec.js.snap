// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ListItem Should correctly save snapshot of the ListItem 1`] = `ShallowWrapper {}`;
